### PR TITLE
fix: trtllm_guide.md

### PR DESCRIPTION
**Pull request by prnake**
Based on https://github.com/triton-inference-server/server/issues/6813#issuecomment-1905106294 and https://github.com/triton-inference-server/tutorials/pull/73, outdated documents may cause misleadings.

### CODE REVIEW

**Comment by dyastremsky** [comment]
Thank you for submitting this pull request! Have you contributed a [Contributor License Agreement](https://github.com/triton-inference-server/server/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) yet? If not, could you?

We can then review the pull request.

**Comment by prnake** [comment]
> Thank you for submitting this pull request! Have you contributed a [Contributor License Agreement](https://github.com/triton-inference-server/server/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) yet? If not, could you?
> 
> We can then review the pull request.

Sure, I've contributed the CLA.

**Comment by dyastremsky** [comment]
Thanks for submitting the CLA! We'll review the CLA shortly.

@jbkyang-nvi @pskiran1, would you be able to review this PR or recommend who would be a good reviewer?

File: Popular_Models_Guide/Llama2/trtllm_guide.md [hunk 1]
```diff
@@ -126,19 +126,21 @@ To run our Llama2-7B model, you will need to:
...
+    MAX_BATCH_SIZE=64
+    python3 ${FILL_TEMPLATE_SCRIPT} -i ${MODEL_FOLDER}/preprocessing/config.pbtxt tokenizer_dir:${TOKENIZER_DIR},tokenizer_type:llama,triton_max_batch_size:${MAX_BATCH_SIZE},preprocessing_instance_count:1
+    python3 ${FILL_TEMPLATE_SCRIPT} -i ${MODEL_FOLDER}/postprocessing/config.pbtxt tokenizer_dir:${TOKENIZER_DIR},tokenizer_type:llama,triton_max_batch_size:${MAX_BATCH_SIZE},postprocessing_instance_count:1
+    python3 ${FILL_TEMPLATE_SCRIPT} -i ${MODEL_FOLDER}/tensorrt_llm_bls/config.pbtxt triton_max_batch_size:${MAX_BATCH_SIZE},decoupled_mode:False,bls_instance_count:1,accumulate_tokens:False
+    python3 ${FILL_TEMPLATE_SCRIPT} -i ${MODEL_FOLDER}/ensemble/config.pbtxt triton_max_batch_size:${MAX_BATCH_SIZE}
+    python3 ${FILL_TEMPLATE_SCRIPT} -i ${MODEL_FOLDER}/tensorrt_llm/config.pbtxt triton_max_batch_size:${MAX_BATCH_SIZE},decoupled_mode:False,max_beam_width:1,engine_dir:${ENGINE_DIR},max_tokens_in_paged_kv_cache:2560,max_attention_window_size:2560,kv_cache_free_gpu_mem_fraction:0.5,exclude_input_in_output:True,enable_kv_cache_reuse:False,batching_strategy:inflight_fused_batching,max_queue_delay_microseconds:600
```
Comment (jbkyang-nvi) [inline Popular_Models_Guide/Llama2/trtllm_guide.md:142]: Are these config.pbtxt parameters all needed?

**Review by jbkyang-nvi — COMMENTED** [review]
Thanks for the change. Had some questions

File: Popular_Models_Guide/Llama2/trtllm_guide.md (hunk 1 above)
Comment (prnake) [reply Popular_Models_Guide/Llama2/trtllm_guide.md:142]: Most of the parameters comes from https://github.com/triton-inference-server/tutorials/pull/73, the doc before the change was missing several key parameters that allow triton to start properly.

Comment (jbkyang-nvi) [reply Popular_Models_Guide/Llama2/trtllm_guide.md:142]: I faced some issues with #73 's changes. Does this work for you?

Comment (prnake) [reply Popular_Models_Guide/Llama2/trtllm_guide.md:142]: I'm testing it with llama2-70b and this change works for me, and I think using the provided FILL_TEMPLATE_SCRIPT is a better method than sed.

**Comment by prnake** [comment]
I checked my logs and found that the main issue with the current template is the lack of `max_batch_size`, which causes an error when Triton tries to load the model.
Image

**Comment by jbkyang-nvi** [comment]
> I checked my logs and found that the main issue with the current template is the lack of `max_batch_size`, which causes an error when Triton tries to load the model. Image

Thanks for investigating this! Since in the engine creating script helps triton autocomplete a lot of the model configuration parameters when it runs, I would prefer if we only modify the ones that matter. 
Although I agree, we should use the `FILL_TEMPLATE_SCRIPT `. Can you remove the extra parameter specification in your changes? 🙏

**Comment by benash** [comment]
These changes seem to improve upon the currently out-of-date steps. Can we approve and merge?

**Comment by prnake** [comment]
The previous config set `tokenizer_dir, tokenizer_type, decoupled_mode, engine_dir`, and I've edit the commit with following changes:
1. using `fill_template.py` script
2. set `tokenizer_dir, tokenizer_type, decoupled_mode, engine_dir` and extra `triton_max_batch_size`
3. add some documents based on my use

**Comment by jbkyang-nvi** [comment]
@prnake  can you fill out https://github.com/triton-inference-server/server/blob/main/CONTRIBUTING.md please? Sorry for the long delay from our side

**Comment by jbkyang-nvi** [comment]
closes https://github.com/triton-inference-server/server/issues/6813

**Pull request merged**